src/Mod/RuleSoldier.cpp

```diff
@@ -95,6 +95,7 @@ void RuleSoldier::load(const YAML::Node &node, Mod *mod, int listOrder, const Mo
 	}
 	_dogfightExperience.merge(node["dogfightExperience"].as<UnitStats>(_dogfightExperience));
 	_armor = node["armor"].as<std::string>(_armor);
+	_specWeapon = node["specialWeapon"].as<std::string>(_specWeapon);
```

Comment:
I think better is not use string but `const RuleItem*` and use `afterLoadHelper("research", this, _research, &RuleResearch::afterLoad);` function like that to link final object rule

Reply:
I used the same way that this is done for the units and armors, or so I thought.
What does this do? `afterLoadHelper("research", this, _research, &RuleResearch::afterLoad);`

Reply:
This is only one case of more general functionality, you should look what exactly this function do. It link and check if all values read from yaml are correct and link correctly to other rule objects